flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecWindowJoin.java

```diff
@@ -0,0 +1,179 @@
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.Lists;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** {@link StreamExecNode} for WindowJoin. */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class StreamExecWindowJoin extends ExecNodeBase<RowData>
+        implements StreamExecNode<RowData>, SingleTransformationTranslator<RowData> {
+    public static final String FIELD_NAME_JOIN_SPEC = "joinSpec";
+    public static final String FIELD_NAME_LEFT_WINDOWING = "leftWindowing";
+    public static final String FIELD_NAME_RIGHT_WINDOWING = "rightWindowing";
+
+    @JsonProperty(FIELD_NAME_JOIN_SPEC)
+    private final JoinSpec joinSpec;
+
+    @JsonProperty(FIELD_NAME_LEFT_WINDOWING)
+    private final WindowingStrategy leftWindowing;
+
+    @JsonProperty(FIELD_NAME_RIGHT_WINDOWING)
+    private final WindowingStrategy rightWindowing;
+
+    public StreamExecWindowJoin(
+            JoinSpec joinSpec,
+            WindowingStrategy leftWindowing,
+            WindowingStrategy rightWindowing,
+            InputProperty leftInputProperty,
+            InputProperty rightInputProperty,
+            RowType outputType,
+            String description) {
+        this(
+                joinSpec,
+                leftWindowing,
+                rightWindowing,
+                getNewNodeId(),
+                Lists.newArrayList(leftInputProperty, rightInputProperty),
+                outputType,
+                description);
+    }
+
+    @JsonCreator
+    public StreamExecWindowJoin(
+            @JsonProperty(FIELD_NAME_JOIN_SPEC) JoinSpec joinSpec,
+            @JsonProperty(FIELD_NAME_LEFT_WINDOWING) WindowingStrategy leftWindowing,
+            @JsonProperty(FIELD_NAME_RIGHT_WINDOWING) WindowingStrategy rightWindowing,
+            @JsonProperty(FIELD_NAME_ID) int id,
+            @JsonProperty(FIELD_NAME_INPUT_PROPERTIES) List<InputProperty> inputProperties,
+            @JsonProperty(FIELD_NAME_OUTPUT_TYPE) RowType outputType,
+            @JsonProperty(FIELD_NAME_DESCRIPTION) String description) {
+        super(id, inputProperties, outputType, description);
+        checkArgument(inputProperties.size() == 2);
+        this.joinSpec = checkNotNull(joinSpec);
+        this.leftWindowing = leftWindowing;
+        this.rightWindowing = rightWindowing;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected Transformation<RowData> translateToPlanInternal(PlannerBase planner) {
+        // validate window strategy
+        if (!leftWindowing.isRowtime() || !rightWindowing.isRowtime()) {
```

Comment:
I think these validation can be finished in the constructor of this class, right? You can extract a validate method.